disnake/ext/tasks/__init__.py

```diff
@@ -551,21 +551,24 @@ def _get_next_sleep_time(self) -> datetime.datetime:
             if self._current_loop == 0:
                 # if we're at the last index on the first iteration, we need to sleep until tomorrow
                 return datetime.datetime.combine(
-                    datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=1),
+                    utcnow() + datetime.timedelta(days=1),
                     self._time[0],
                 )
 
         next_time = self._time[self._time_index]
 
         if self._current_loop == 0:
             self._time_index += 1
-            return datetime.datetime.combine(
-                datetime.datetime.now(datetime.timezone.utc), next_time
-            )
+            if next_time > utcnow().timetz():
+                return datetime.datetime.combine(utcnow(), next_time)
+            else:
+                return datetime.datetime.combine(
+                    utcnow() + datetime.timedelta(days=1),
+                    next_time,
+                )
 
-        next_date = self._last_iteration
-        if self._time_index == 0:
-            # we can assume that the earliest time should be scheduled for "tomorrow"
+        next_date = cast(datetime.datetime, self._last_iteration)
```

Comment:
```suggestion
        next_date = self._last_iteration
```
This `cast` isn't required (and it's not being imported anyway)